### PR TITLE
Plotting for perfusion alignment

### LINKED ISSUE #1369: Add plotting functionality in Perfusion Alignment
Add plotting functionality in Perfusion Alignment.

### PR DESCRIPTION
**Pull request by ashishsingh18**
Fix #1369

### CODE REVIEW

File: src/applications/PerfusionAlignment.cpp [hunk 1]
```diff
@@ -16,4 +16,34 @@ See COPYING file or https://www.med.upenn.edu/cbica/captk/license.html
 
 #include "PerfusionAlignment.h"
 
-
+#include <vtkChartXY.h>
+#include <vtkTable.h>
+#include <vtkFloatArray.h>
+#include <vtkPlot.h>
+
+void PerfusionAlignment::Createchart(vtkChartXY* chart, std::map<std::string, std::vector<float>*>* map)
```
Comment (sarthakpati) [inline src/applications/PerfusionAlignment.cpp:24]: What is the `std::map` structure?

Comment (sarthakpati) [inline src/applications/PerfusionAlignment.cpp:24]:
```suggestion
void PerfusionAlignment::Createchart(vtkChartXY* chart, std::map<std::string, std::vector<float>> map)
```

File: src/applications/PerfusionAlignment.h [hunk 2]
```diff
@@ -160,6 +161,13 @@ class PerfusionAlignment
     return std::make_pair(stdDevImage, stdDevVector);
   }
 
+  /**
+  \brief fill the chart with supplied perfusion alignment data
+  \param chart pointer to vtkChartXY that will be filled
+  \param map pointer to map of data name to pointer of vector of data that needs to be plotted
+  */
+  void Createchart(vtkChartXY* chart, std::map<std::string, std::vector<float>*>* map);
```
Comment (sarthakpati) [inline src/applications/PerfusionAlignment.h:169]:
```suggestion
  void Createchart(vtkChartXY* chart, std::map<std::string, std::vector<float>> map);
```

**Review by AlexanderGetka-cbica — APPROVED** [review]
This seems good. Re: Sarthak's comments, I think this works either way, though pass by value requires a little more memory. I think the default copy constructor on the map would be shallow, so std::map<std::string, std::vector<float>>* should negate most memory impact while still being very quickly readable.

File: src/applications/PerfusionAlignment.cpp (hunk 1 above)
Comment (ashishsingh18) [reply src/applications/PerfusionAlignment.cpp:24]: No change needed. What we have is correct and avoids duplication of memory.

File: src/applications/PerfusionAlignment.h (hunk 2 above)
Comment (ashishsingh18) [reply src/applications/PerfusionAlignment.h:169]: No change needed. What we have is correct and avoids duplication of memory.

**Comment by ashishsingh18** [comment]
> 
> 
> This seems good. Re: Sarthak's comments, I think this works either way, though pass by value requires a little more memory. I think the default copy constructor on the map would be shallow, so `std::map<std::string, std::vector<float>>*` should negate most memory impact while still being very quickly readable.

Yeah, no change is needed. What we have is correct and avoids duplication of memory.

**Review by sarthakpati — APPROVED** [review]

**Comment by sarthakpati** [comment]
> > This seems good. Re: Sarthak's comments, I think this works either way, though pass by value requires a little more memory. I think the default copy constructor on the map would be shallow, so `std::map<std::string, std::vector<float>>*` should negate most memory impact while still being very quickly readable.
> 
> Yeah, no change is needed. What we have is correct and avoids duplication of memory.

Alright. The changes make sense; approved.

**Pull request merged**